File: Ejercicios Propuestos 3/Ejercicio 22/Ejercicio 22.jl
function InvertirNumero(num)
    
    
    invertido = 0
    esNegativo = false
    
    # Verificamos si el número es negativo
    if num < 0
        esNegativo = true
        num = num * (-1) 
    end
    
    # Proceso inversión
    while num > 0
        digito = num % 10
        invertido = (invertido * 10) + digito
        num = trunc(Int, num / 10)
    end
    
    # Si originalmente era negativo, le devolvemos el signo al resultado
    if esNegativo
        invertido = invertido * (-1) 
    end
    
    return invertido
end


print("Ingrese un número entero (puede ser negativo): ")
numeroOriginal = parse(Int, readline())

# función que devuelve el valor calculado
resultado = InvertirNumero(numeroOriginal)

println("----------------------")
println("Número original: ", numeroOriginal)
println("Número invertido: ", resultado)

File: Ejercicios Propuestos 3/Ejercicio 22/ejercicio22.cpp
#include <iostream>
using namespace std;

int invertirNumero(int n) {
    int invertido = 0;
    while (n > 0) {
        int digito = n % 10;
        invertido = invertido * 10 + digito;
        n = n / 10;
    }
    return invertido;
}

int main() {
    int numero;
    cout << "Ingrese un numero entero: ";
    cin >> numero;

    cout << "Numero invertido: " << invertirNumero(numero) << endl;
    return 0;
}
